zap/src/main/java/org/zaproxy/zap/view/ContextIncludePanel.java

```diff
@@ -64,6 +75,56 @@ public static String getPanelName(Context context) {
         return getPanelName(context.getId());
     }
 
+    private File selectImportFile() {
+        JFileChooser chooser = new JFileChooser(Constant.getContextsDir());
+        File file = null;
+
+        int rc = chooser.showOpenDialog(View.getSingleton().getMainFrame());
+        if (rc == JFileChooser.APPROVE_OPTION) {
+            file = chooser.getSelectedFile();
+            if (file == null || !file.exists()) {
+                return null;
+            }
+            return file;
+        }
+        return null;
+    }
+
+    private void addImportButton() {
+        String IMPORT_BUTTON_LABEL = Constant.messages.getString("context.include.import.label");
+        JButton importButton = new JButton(IMPORT_BUTTON_LABEL);
+        importButton.addActionListener(
+                new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent evt) {
+                        File file = selectImportFile();
+                        if (file != null) {
+                            List<String> regexes = regexesPanel.getRegexes();
+
+                            String line = null;
+                            try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+                                while ((line = br.readLine()) != null) {
+                                    String trimmedRegex = line.trim();
+                                    if (!regexes.contains(trimmedRegex)) {
+                                        Pattern.compile(trimmedRegex, Pattern.CASE_INSENSITIVE);
+                                        regexes.add(trimmedRegex);
+                                    }
+                                }
+                            } catch (IOException | PatternSyntaxException e) {
+                                View.getSingleton()
+                                        .showWarningDialog(
+                                                Constant.messages.getString(
+                                                        "context.include.import.error",
+                                                        e.getLocalizedMessage()));
```

Comment:
This has bitten us before. I'd suggest including two substitutions. One that is specifically the failed pattern string, and one that's the message or localizedMessage.

Edit: To be more specific, localizedMessage doesn't always exist, or can sometimes be null (I forget precisely if we hit null or simply empty in the past).

Reply:
Thanks for the hint. Changed this to get the error detail (e.getLocalizedMessage) separately. Checking for null before passing it to string replacement to avoid failure. If the message is empty it is not a big deal, in this case just the main error message is displayed without any detail.